addons/mail/static/src/model/model_manager.js

```diff
@@ -431,25 +430,49 @@ export class ModelManager {
     }
 
     /**
+     * Adds fields, methods, and other stuff from the model definition to the
+     * Model.
+     *
      * @private
-     * @param {mail.model} Model class
-     * @param {Object} patch
+     * @param {mail.model} Model
      */
-    _applyModelPatchFields(Model, patch) {
-        for (const [fieldName, field] of Object.entries(patch)) {
-            Model.fields[fieldName] = field;
+    _applyModelDefinition(Model) {
+        const definition = registry[Model.name];
+        Object.assign(Model, definition.modelMethods);
+        Object.assign(Model.prototype, definition.recordMethods);
+        // use `modelName` rather than `name` to avoid conflicting with the name
+        // inherited from Function
```

Comment:
I don't think so, but I haven't actually tried. I was going to do it until I realised the diff it would make, then I figured that using the same identifier as a property already taken on the prototype sounded like a terrible idea anyway, no matter how much diff it makes.

Reply:
Then this is a bad comment. It looks like the main reason is just that we want a different name (and/or avoid diff), not to actually avoid a conflict.

Reply:
I just checked and it looks like `Model.name` actually contains exactly this value already, so this is effectively a duplicate value and a non-issue.

You use `Model.name` 3 lines above when reading the registry.